backend/api/v1/endpoints/connections.py

```
@@ -0,0 +1,48 @@
+def reject_connection(request_id: int, db: Session = Depends(get_db), current_user: dict = Depends(get_current_user)):
+    connection = reject_request(db, request_id)
+    if not connection:
+        raise HTTPException(status_code=404, detail="No pending request found.")
+    return {"message": "Connection rejected!"}
+
+@router.get("/connections")
+def list_connections(db: Session = Depends(get_db), current_user: dict = Depends(get_current_user)):
+    return get_connections(db, current_user["id"])
+
+@router.get("/users")
+def get_users(current_user: dict = Depends(get_current_user), db: Session = Depends(get_db)) -> list[User]:
```

Ruff (0.8.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)